### PR TITLE
For EXP-2992: Add notification config with polling interval to messaging feature

### PR DESCRIPTION
**Pull request by jhugman**
Fixes [EXP-2992](https://mozilla-hub.atlassian.net/browse/EXP-2992).

Adds a polling interval to the messaging feature.

This can be accessed via 

```kotlin
val featureConfig = FxNimbus.features.messaging.value()
val notificationConfig = featureConfig.notificationConfig

val pollingInterval = notificationConfig.pollingInterval
```

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

File: .experimenter.yaml
Comment (t-p-white) [inline .experimenter.yaml:42]: does this also require a default as nimbus.fml.yaml

Comment (t-p-white) [inline .experimenter.yaml:42]: nit:
```suggestion
      description: "Configuration of the notification worker for all notification messages."
```

File: nimbus.fml.yaml
Comment (t-p-white) [inline nimbus.fml.yaml:354]: should this also match the default 3600?

File: nimbus.fml.yaml
Comment (t-p-white) [inline nimbus.fml.yaml:188]: is the new `channel: developer` notation required to add the value, or can this be included in the above `channel: developer`?

**Review by t-p-white — CHANGES_REQUESTED** [review]
Great description 👍  

Newbie question - other PRs I’ve seen & created for mobile use a different branching process (like this https://github.com/mozilla-mobile/shared-docs/blob/main/android/CONTRIBUTING_code.md#creating-a-pull-request) whereas this is using a branch directly in the fenix repo - just want to clarify if this is deliberate?

**Review by t-p-white — APPROVED** [review]
lgtm 👍 

Shall we hold off landing until we get some feedback on the branching query?

**Comment by MatthewTighe** [comment]
> lgtm 👍
> 
> Shall we hold off landing until we get some feedback on the branching query?

I've seen other people that have access levels to create branches in the repo follow this branching strategy before, for [example](https://github.com/mozilla-mobile/fenix/pull/27993) so AFAIK it's okay, but do not have an good understanding of the difference so I may not be an authority.

**Comment by Mugurell** [comment]
> > lgtm 👍
> > Shall we hold off landing until we get some feedback on the branching query?
> 
> I've seen other people that have access levels to create branches in the repo follow this branching strategy before, for [example](https://github.com/mozilla-mobile/fenix/pull/27993). So AFAIK it's okay, but I do not have an good understanding of the difference so I may not be an authority.

I think this branches will remain in the project even after the patch is merged, polluting the release branches list.
I remember @rvandermeulen cleaning these from time to time, maybe he can offer more context.

**Comment by t-p-white** [comment]
> > > lgtm 👍
> > > Shall we hold off landing until we get some feedback on the branching query?
> > 
> > 
> > I've seen other people that have access levels to create branches in the repo follow this branching strategy before, for [example](https://github.com/mozilla-mobile/fenix/pull/27993). So AFAIK it's okay, but I do not have an good understanding of the difference so I may not be an authority.
> 
> I think this branches will remain in the project even after the patch is merged, polluting the release branches list. I remember @rvandermeulen cleaning these from time to time, maybe he can offer more context.

Thank you @MatthewTighe and @Mugurell for your thoughts on this! I'll see if @rvandermeulen can give some steer. It'd be useful to know for future reference 👍

**Comment by rvandermeulen** [comment]
AFAIK, mergify should delete the branch automatically when the PR merges. Worst-case, we can always remove it manually if it doesn't, though.

**Comment by t-p-white** [comment]
> AFAIK, mergify should delete the branch automatically when the PR merges. Worst-case, we can always remove it manually if it doesn't, though.

Super, thank you for the info @rvandermeulen!

File: nimbus.fml.yaml
Comment (t-p-white) [inline nimbus.fml.yaml:189]: Can I clarify why this is required to be 3 minutes? Reason for asking is that typically Workers are restricted by the OS with regard to how often they can be performed

**Pull request merged**
